core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1889,9 +1889,13 @@ object SnappySession extends Logging {
     def apply(session: SnappySession, lp: LogicalPlan, sqlText: String, pls:
     ArrayBuffer[ParamLiteral]): CachedKey = {
 
+      val invalidate = Array(false)
```

Comment:
Should be private. Also I think what Vivek meant was why not "private var invalidate = false" instead of using an array? A var is certainly preferred in this case instead of an array to change value.